### DISCUSSION TITLE
Search memory for string data

### DISCUSSION THREAD

**Discussion by meleu** [RetroAchievements/RAIntegration#759] [Ideas]
_From @Jamiras on March 21, 2018 15:12_

As a developer, I'd like to be able to search memory for string patterns. As I may not know the exact byte->tile mapping used by the game, I'd like to be able to enter a string and have the toolkit look for patterns in the memory that could imply the tile mapping.

For example, searching for "25000" would match "C2 C5 C0 C0 C0" or "06 09 04 04 04" as the second byte is 3 bytes higher than the first, and the last three are the same and two bytes lower than the first.

_Copied from original issue: RetroAchievements/RASuite#62_

**Comment by Jamiras** [comment]
ASCII search was added via #609. It could still be useful to search for text-like patterns.

**Comment by televandalist** [comment]
What about a config for custom character encoding? 

For example: when A-Z = 0x45-0x5E and a-z = 0x5F-0x78, as opposed to ASCII's A-z = 0x41-0x7A...

It's something I run into quite a bit so it would be neat to be able to switch between ASCII and custom 🤔
